### DISCUSSION TITLE
request.signal is never aborted

### DISCUSSION THREAD

**Discussion by adaboese** [remix-run/remix#8461] [General]
I am trying to abort SSR when the original request is aborted, i.e.

```ts
export const loader = async ({ params, request }: LoaderFunctionArgs) => {
  return eventStream(request.signal, function setup(send) {
    const timer = setInterval(() => {
      console.log('TIME', request.signal.aborted);

      send({ data: new Date().toISOString(), event: 'time' });
    }, 1_000);

    return function clear() {
      console.log('CLEAR');

      clearInterval(timer);
    };
  });
};
```

But I just realized that `CLEAR` is never called and `request.signal.aborted` is always `false` – even after I navigate away from the page.

For what it is worth, I think this happens only is Vite dev server and it works as expected in production.

**Comment by lukasa1993** [comment]
vite dev has issues, if you use express template it works under express

**Comment by hi-ogawa** [comment]
Interesting. The difference between dev (default builtin vite dev server) and prod (remix-serve express server) might be caused by this PR:
- https://github.com/remix-run/remix/pull/8130

These two are mostly same except that the PR removed this part since it had an unintended effect on the usage of React `renderToReadableStream` on Node:

https://github.com/remix-run/remix/blob/4968e3a82924d1b964ef19552ae6fa4e11c43e3b/packages/remix-express/server.ts#L99-L101

Now looking back the history of the code around `AbortController`, it looks like it wasn't that simple, for example, there is some platform-specific aspect of things discussed in:
- https://github.com/remix-run/remix/pull/3626

(Actually I thought `req.on("close", ...)` is preferred over `res.on("close", ...)`, but apparently the exact opposite happened in the PR.)

I think it's worth raising an issue. If you have a repro, then that would be great as well.

---

Probably my PR was premature (namely https://github.com/remix-run/remix/pull/8062 and https://github.com/remix-run/remix/pull/8130). I didn't check at that time, but there is some difference left in Response handling and the actual fix should probably take into account this part as well. Also it's totally possible that react-dom `renderToReadableStream` on NodeJs might be mis-used.
 
https://github.com/remix-run/remix/blob/4968e3a82924d1b964ef19552ae6fa4e11c43e3b/packages/remix-express/server.ts#L117

https://github.com/remix-run/remix/blob/4968e3a82924d1b964ef19552ae6fa4e11c43e3b/packages/remix-dev/vite/node/adapter.ts#L57-L59

**Comment by DanielBaulig** [comment]
Running into this exact same issue. Was there ever an issue filed? I'll open one if not.

**Reply by hi-ogawa** [reply]
I don't think it's been raised yet. It would be great if you can create one.
Also now that `toNodeRequest` is also used for cloudflare integration, so you can check how `AbortController` goes with `react-dom/server`'s renderToReadableStream

https://github.com/remix-run/remix/blob/2e65318b15df407b31b101362e03729e61ec5622/packages/remix-dev/vite/cloudflare-proxy-plugin.ts#L73

**Comment by DanielBaulig** [comment]
I crated a stackblitz repro and filed a bug: https://github.com/remix-run/remix/issues/9438

**Comment by brophdawg11** [comment]
This should be resolved via https://github.com/remix-run/remix/pull/9976

**Comment by madfcat** [comment]
```
        request.signal.addEventListener("abort", () => {
          console.log("Request aborted by client");
          controller.error(new DOMException("Aborted", "AbortError"));
        });
```
        
Not working for me inside `ReadableStream`.
     
     
       
Using:

```
    "@remix-run/dev": "^2.16.1",
    "@remix-run/fs-routes": "^2.16.1",
    "@remix-run/node": "^2.16.1",
    "@remix-run/react": "^2.16.1",
    "@remix-run/serve": "^2.16.1",
```
